client/providers/CallProvider/hooks/useVoipClient.ts

```diff
@@ -0,0 +1,82 @@
+import { useSafely } from '@rocket.chat/fuselage-hooks';
+import { useEffect, useMemo, useState } from 'react';
+
+import { IRegistrationInfo } from '../../../../definition/voip/IRegistrationInfo';
+import { useEndpointData } from '../../../hooks/useEndpointData';
+import { AsyncStatePhase } from '../../../lib/asyncState';
+import { SimpleVoipUser } from '../../../lib/voip/SimpleVoipUser';
+import { VoIPUser } from '../../../lib/voip/VoIPUser';
+import { useWebRtcServers } from './useWebRtcServers';
+
+type UseVoipClientResult =
+	| UseVoipClientResultResolved
+	| UseVoipClientResultError
+	| UseVoipClientResultLoading;
+
+type UseVoipClientResultResolved = {
+	voipClient: VoIPUser;
+	registrationInfo: IRegistrationInfo;
+};
+type UseVoipClientResultError = { error: Error };
+type UseVoipClientResultLoading = Record<string, never>;
+
+export const isUseVoipClientResultError = (
+	result: UseVoipClientResult,
+): result is UseVoipClientResultError => !!(result as UseVoipClientResultError).error;
+
+export const isUseVoipClientResultLoading = (
+	result: UseVoipClientResult,
+): result is UseVoipClientResultLoading => Object.keys(result).length === 0;
+
+export const useVoipClient = (): UseVoipClientResult => {
+	const config = useEndpointData(
+		'connector.extension.getRegistrationInfo',
+		useMemo(() => ({ extension: '80000' }), []),
+	);
+
+	// TODO: should we recreate the client if the server list changes? this can disrupt the call
```

Comment:
?

Reply:
This refers to 

`const iceServers = useWebRtcServers();`

if the servers change that could disrupt a call, most of these todos are things to think about when doing the next steps